render/renderable_node.go

```diff
@@ -107,6 +70,73 @@ func newDerivedPseudoNode(id, major string, node RenderableNode) RenderableNode
 		Pseudo:       true,
 		Origins:      node.Origins.Copy(),
 		EdgeMetadata: node.EdgeMetadata.Copy(),
-		NodeMetadata: report.MakeNodeMetadata(),
+		NodeMetadata: node.NodeMetadata.Copy(),
+	}
+}
+
+// WithNodeMetadata creates a new RenderableNode based on rn, with n
+func (rn RenderableNode) WithNodeMetadata(n report.NodeMetadata) RenderableNode {
+	result := rn.Copy()
+	result.NodeMetadata = n
```

Comment:
Yes why not.  Its only used in constructors, so it should affect anything, but this would be nicer.